src/GraphQL/Utilities/AstPrinter.cs

```diff
@@ -328,25 +336,54 @@ public AstPrintVisitor()
                 c.Field(x => x.Value);
                 c.Print(f =>
                 {
-                    var val = f.Arg(x => x.Value);
-                    if (!string.IsNullOrWhiteSpace(val?.ToString()) && !val.ToString().StartsWith("\"", StringComparison.InvariantCulture))
+                    var val = (string)f.Arg(x => x.Value);
+                    if (val == null)
+                        throw new InvalidOperationException("StringValue cannot contain null");
```

Comment:
Here is the same situation. This check is not the responsibility of `AstPrinter`. Let's move it into `StringValue` ctor.